test.py

```diff
@@ -0,0 +1,42 @@
+import torch
+from vit_pytorch import MPP, ViT
+# from vit_pytorch import ViT, MaskedPredictionLoss
+
+v = ViT(
+    image_size = 256,
+    patch_size = 32,
+    num_classes = 3,
```

Comment:
so my understanding was that for each of the channels (r, g, b), they found the average, and then discretize to 3 bit, which would yield 9 bits, and thus be predicting 512 classes here

but correct me if i'm wrong!

Reply:
I'm not sure if this logic works, but I was thinking that I can just keep it as 3 channel, but have each channel be discretized to 0, 1, or 2 to replicate the 3 bits for each channel

Reply:
Its probably just more straightforward and avoids any confusion to just discretize to 3 bits for each channel though

Reply:
yup, just have each channel discretize to 0, 1, 2, then add one, cumulative product, subtract one to get it into the range of 0 - 512

Reply:
i don't think that works, i think you'll want to discretize to 8 values for each channel, then do a cumulative product of the three channels to get it into the range of 0 - 512